STL_Extension/include/CGAL/Concurrent_compact_container.h

```diff
@@ -341,17 +341,25 @@ class Concurrent_compact_container
   iterator
   emplace(const Args&... args)
   {
+    typedef CCC_internal::Erase_counter_strategy<
+      CCC_internal::has_increment_erase_counter<T>::value> EraseCounterStrategy;
     FreeList * fl = get_free_list();
     pointer ret = init_insert(fl);
+    auto erase_counter = EraseCounterStrategy::erase_counter(*ret);;
```

Comment:
Note that we still use an undefined behavior, because:
  - here the code reads the erase counter from `*ret`, that is not yet created,
  - in `allocator_new_block`, the erase counter of uninitialized objects is assigned.

Reply:
Should we try to change that now, instead of waiting until it breaks and spend another 2 days trying to understand why ?

Reply:
As we use bit-squatting for the type `USED`/`FREE` of a cell of the container, we have to use undefined-behavior, no matter what we do. That is a property of that container: it uses undefined-behavior for efficiency. That is worth a few days of hard work from time to time.

Reply:
Ok, I was right to ask then :)